src/app/administrator/admin-publisher/admin-publisher.component.ts

```diff
@@ -1,15 +1,63 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit } from "@angular/core";
+import { OLmsService } from "src/app/common/o/services/oLms.service";
+import { environment } from "src/environments/environment";
 
 @Component({
-  selector: 'app-admin-publisher',
-  templateUrl: './admin-publisher.component.html',
-  styleUrls: ['./admin-publisher.component.css']
+  selector: "app-admin-publisher",
+  templateUrl: "./admin-publisher.component.html",
+  styleUrls: ["./admin-publisher.component.css"],
 })
 export class AdminPublisherComponent implements OnInit {
+  publishers: any[];
+  books: any[];
 
-  constructor() { }
+  constructor(private lmsService: OLmsService) {}
 
   ngOnInit() {
+    this.readPublishers();
+    this.readBooks();
   }
 
+  readPublishers() {
+    this.lmsService
```

Comment:
I pushed the functions that talk to the database to the service class and removed the business logic. I would like to talk to Pramod about exactly how to determine what goes in Service vs. Borrower.

Reply:
I remember he said that generally business logic goes in the component layer and not the service layer for Angular apps. But in his example code he also wrote a service to paginate the table. My inference is that business logic should generally go in components, unless it clearly makes sense to have it in a service instead because multiple components will want to use it as-is.